ffi/src/handle.rs

```diff
@@ -186,20 +270,18 @@ pub trait ArcHandle: Sized {
     }
 }
 
-/// A special kind of [ArcHandle] which is optimized for [Sized] types. Handles for sized types are
-/// more efficient if they implement this trait instead of [ArcHandle].
-pub trait SizedArcHandle: Sized {
-    type Target: Sized;
-}
+/// A marker trait that allows a [Sized] type to act as its own [ArcHandle]. This is more efficient
+/// and also produces a pointer of the same type that can be used directly (no type aliasing).
+pub trait SizedArcHandle: Sized { }
 
 // A blanket implementation of `ArcHandle` for all types satisfying `SizedArcHandle`.
 //
-// Unlike the default (unsized) implementation, which must wrap the Arc in a Box in order to obtain
-// a thin pointer, the sized implementation can directly return a pointer to the Arc's underlying
+// Unlike the default (unsized) implementation, which must wrap the `Arc` in a `Box` in order to
+// obtain a thin pointer, the sized implementation can directly return a pointer to the underlying
```

Comment:
Would we want both `[]`and backticks? Or just `[]`? Seems like the latter?

Also, I wonder if we _want_ links for every single mention of every class? The std rust docs don't seem overly "linky"? e.g. the doc comment for [Box::pin](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.pin) doesn't actually link to [Pin](https://doc.rust-lang.org/std/pin/struct.Pin.html), presumably because the method signature already did so. 

Anyway, I'll take a pass and be intentional about linking vs. merely backticking.

Reply:
I usually do [`Foo`] because it makes it a link and puts it in `fixed width`, which is also what the std docs do.

I agree we don't need to link _everything_, but here this was the sort of "intro" doc to the type so it seems like it would make sense to give links to the underlying types involved.